### PR TITLE
Ensure that the various Min and Max APIs are accelerated where possible

### PR DESCRIPTION
**Pull request by tannergooding**
This mostly does AVX-512 based acceleration but it also ensures that these are supported on Arm64 where that's trivial to do.

### CODE REVIEW

File: src/coreclr/jit/importercalls.cpp [hunk 1]
```diff
@@ -3398,161 +3398,86 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
...
             }
-#if defined(TARGET_ARM64)
-            // ARM64 has fmax/fmin which are IEEE754:2019 minimum/maximum compatible
-            case NI_System_Math_Max:
-            case NI_System_Math_Min:
-#endif
 
-#if defined(FEATURE_HW_INTRINSICS) && defined(TARGET_XARCH)
             case NI_System_Math_Max:
-            case NI_System_Math_Min:
```
Comment (tannergooding) [inline src/coreclr/jit/importercalls.cpp:3409]: This logic was all extracted to a new helper function `impMinMaxIntrinsic`, to avoid code duplication.

Comment (tannergooding) [reply src/coreclr/jit/importercalls.cpp:3409]: Doing so made the diff unhappy, and so it's showing a bunch of code as "new" where it actually isn't

**Comment by tannergooding** [comment]
CC. @dotnet/jit-contrib, @dotnet/avx512-contrib 

As per the top post this ensures various Min/Max APIs are accelerated using available instructions on both Arm64 and x86/x64. For the latter it utilizes AVX-512 where possible.

[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=310154&view=ms.vss-build-web.run-extensions-tab) look good with us saving 16k bytes of codegen on Linux x64 and 29k bytes of codegen on Windows x64 (significantly simpler code that is now branch-free).

Cleaning up the intrinsic recognition path also brings a `-0.12%` TP improvement for ASP.NET and some smaller improvements for other scenarios.

File: src/coreclr/jit/importercalls.cpp [hunk 2]
```diff
@@ -3398,161 +3398,86 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
...
-                    op2 = otherNode;
-                }
-                else
-                {
-                    assert(ni == NI_System_Math_Min);
+            case NI_System_Math_Min:
+            {
+                bool isMax       = false;
+                bool isMagnitude = false;
+                bool isNumber    = false;
```
Comment (BruceForstall) [inline src/coreclr/jit/importercalls.cpp:3416]: nit: these (and similarly, elsewhere) could all be `constexpr` or `const`

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**